unity-client/Assets/Scripts/MainScripts/DCL/Components/Audio/Tests/AudioTests.cs

```diff
@@ -220,6 +221,61 @@ public IEnumerator AudioClipAttachedGetsReplacedOnNewAttachment()
                 scene, CLASS_ID.AUDIO_CLIP);
         }
 
+        [UnityTest]
+        public IEnumerator VolumeIsToggledWhenCharacterEntersOrLeavesScene()
```

Comment:
Maybe not for this PR but this suite requires a huge refactor. Most of tests are monolithic and testing multiple features at once.

In any case, I would try to minimize the future problem converting this new integration test into a unit tests by:
- Moving the audio setup to a `Initialize` method that could be easily converted to a `SetUp` one in the future.
- If `DCLAudioSource` allows it: removing the `AudioClip` related code, in this tests we are testing the volume, we dont care if there's a clip or not. Chances are that we are aborting the process if an `AudioClip` was not found so maybe is not possible.
- Removing all the Teleport code. `DCLAudioSource` react to changes in the `SceneId` SO and that's what we should test here. It will reduce the boilerplate and make the test more reliable, fast and readable.

I would also suggest splitting this test into 4 different cases
- Volume is 0 when DCLAudioSource is created and player is not the scene
- Volume is `model.Volume` when DCLAudioSource is created and player is in the scene
- Volume is 0 when the player moves away from the scene (manually set Volume to anything different)
- Volume is 1 when the player moves into the scene (manually set Volume to anything different)

Reply:
I separated the test un 4 tests as requested, check it out